src/components/ModalState/Icon/component.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { Testable } from '../../../modules/test-ids';
+import { Variant } from '../styled';
+
+import { DialogSuccess, DialogWarning, DialogDanger, DialogIcon } from './styled';
+
+export type Props = Testable & {
+  variant: Variant;
+  icon: React.ReactNode;
+};
+
+const icons: any = {
```

Comment:
Don't use `any`. If you have trouble with types, just use a function `getIcon({ variant })` instead.